src/services/api/movie-service.js

```diff
@@ -0,0 +1,164 @@
+import HttpBaseClient from "../http-base-client";
+
+const ENDPOINTS = {
+    CREATE: "api/movies/store",
+    INDEX: "api/movies/index",
+    SHOW: "api/movies/show/",
+    INC_NUM_VISITS: "api/movies/visit/",
+    COMMENTS: "api/movies/comments/",
+    CREATE_COMMENT: "api/comments/store",
+    POST_REACTION: "api/reactions/store",
+    FAVORITES_BY_USER: "api/favorites/index/user/",
+    ADD_TO_FAVORITES: "api/favorites/store",
+    REMOVE_FROM_FAVORITES: "api/favorites/destroy",
+    UPDATE_FAVORITE: "api/favorites/update",
+    BEST: "api/movies/best",
+    RELATED: "api/movies/genre/",
+};
+
+class MovieService extends HttpBaseClient {
+    createMovie = async (movieData) => {
+        const { title, description, imageUrl, genre } = movieData;
+
+        const newMovie = this.getApiClient()
+            .post(ENDPOINTS.CREATE, {
+                title,
+                description,
+                imageUrl,
+                genre,
+            })
+            .then((response) => {
+                return response.data;
+            });
+
+        return newMovie;
+    };
+
+    getMovies = async () => {
+        const movies = this.getApiClient()
+            .get(ENDPOINTS.INDEX)
+            .then((response) => response.data);
+
+        return movies;
+    };
+
+    selectMovie = async (id) => {
+        const movie = await this.getApiClient()
+            .get(ENDPOINTS.SHOW + id)
+            .then((response) => response.data);
+
+        localStorage.setItem("selectedMovieId", movie.id);
+        localStorage.setItem("selectedMovieTitle", movie.title);
+        localStorage.setItem("selectedMovieDescription", movie.description);
+        localStorage.setItem("selectedMovieImageUrl", movie.imageUrl);
+        localStorage.setItem("selectedMovieGenre", movie.genre);
+        localStorage.setItem("selectedMovieLikes", movie.likes);
+        localStorage.setItem("selectedMovieDislikes", movie.dislikes);
```

Comment:
Ovo je prekomplikovano poprilicno... Nema sanse da je ovo jedino moguce resenje. Ako vec moras da cuvas u LS-u, zasto ne bi cuvao ceo movie objekat tamo, a ne 2813812 polja odvojeno 🤷🏼‍♂️

Reply:
Da, nasao sam na netu da ne moze u LSu da se cuva objekat i onda sam poceo tako da resavam... Da bi mi sada sinulo da moze da se uradi JSON.stringify... Resio sam to uglavnom.